packages/payment-experience-api/src/api/paytrailOnlinePaymentReturnController.ts

```diff
@@ -100,11 +104,40 @@ export class PaytrailOnlinePaymentReturnController extends AbstractController {
       try {
         const paymentId = query['checkout-stamp']?.toString()
         if (paymentId) {
-          await updateInternalPaymentFromPaytrail({
+          const updatedPayment = await updateInternalPaymentFromPaytrail({
             paymentId: paymentId,
             merchantId: parseMerchantIdFromFirstOrderItem(order),
             namespace: order.namespace,
           })
+
+          // check if this was paid late (KYV-1196)
+          if (paytrailStatus.paymentPaid) {
```

Comment:
Add try catch to log return of these functions and to allow function operating as is.